### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by carllerche**

### CODE REVIEW

File: ci/azure-test-stable.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - bash: |
+      sudo apt-get install gcc-multilib
+      rustup target add $TARGET
+      cargo check --all --exclude tokio-tls --target $TARGET
+      cargo check --tests --all --exclude tokio-tls --target $TARGET
+    condition: eq(variables['agent.os'], 'Linux')
+    env: { TARGET: 'i686-unknown-linux-gnu' }
+    displayName: cross compilation
+
+  - bash: |
```
Comment (rylev) [inline ci/azure-test-stable.yml]: @carllerche The jobs are titled "$OS + Stable", but we're now doing explicit nightly testing. Do we want to have a different job? A different template?

**Comment by rylev** [comment]
@carllerche Everything except for the [deploy part of the travis.yml](https://github.com/tokio-rs/tokio/blob/master/.travis.yml#L115-L136) file is now implemented on Pipelines and the tests are passing. Let me know what you think.

**Comment by carllerche** [comment]
Not critical for merging, but i noticed that the rust minimum version check is happening as part of the stable build. I would probably break it out into a top level build (to run parallel) and only have it run on linux.

**Comment by carllerche** [comment]
Ok, this should be ready for a merge.

**Comment by rylev** [comment]
@carllerche Looks awesome! For some reason Appveyor and Travis builds are still running. I'm guess they need to be turned off for this repo?

**Comment by carllerche** [comment]
@rylev yeah i have to figure out how to disable all that

File: ci/azure-install-rust.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain $RUSTUP_TOOLCHAIN
+      echo "##vso[task.setvariable variable=PATH;]$PATH:$HOME/.cargo/bin"
+    env:
+      RUSTUP_TOOLCHAIN: ${{parameters.rust_version}}
+    displayName: Install rust
+    condition: not(eq(variables['Agent.OS'], 'Windows_NT'))
+
+  # Windows.
+  - script: |
+      echo "windows"
```
Comment (LucioFranco) [inline ci/azure-install-rust.yml]: still needed? I dont see this done on linux/mac

File: ci/azure-rustfmt.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# Check formatting
+- job: ${{ parameters.name }}
+  displayName: Check rustfmt
+  pool:
+    vmImage: ubuntu-16.04
+  steps:
+    - template: azure-install-rust.yml
+      parameters:
+        # Pin the version of Rust in case rustfmt changes.
+        rust_version: 1.32.0
```
Comment (LucioFranco) [inline ci/azure-rustfmt.yml]: I feel we should track stable, in the sense that most users will be on latest stable, so in theory the whole repo should stay somewhat up to date.

**Pull request merged**